### ISSUE TITLE
[cpr]Compiling client application in x64 Windows 10 throws linker error (__CxxFrameHandler4)

### ISSUE THREAD

**Issue by rupeshbhurke** [microsoft/vcpkg#15999]
Hi there!

I installed vcpkg & cpr using following commands.

```
mkdir C:\Dev\vcpkg
cd C:\Dev
git clone https://github.com/Microsoft/vcpkg.git
cd vcpkg
bootstrap-vcpkg.bat
vcpkg integrate install
vcpkg install cpr:x86-windows
vcpkg install cpr:x64-windows
```

I created a Console Application using VS 2015 Community Edition. Copied your main() function. Compiling x86 Debug & Release successfully. Compiling on x64 it gives the following linker errors:
```
1>cpr.lib(cookies.cpp.obj) : error LNK2001: unresolved external symbol __CxxFrameHandler4
1>cpr.lib(proxies.cpp.obj) : error LNK2001: unresolved external symbol __CxxFrameHandler4
1>cpr.lib(util.cpp.obj) : error LNK2001: unresolved external symbol __CxxFrameHandler4
1>cpr.lib(cprtypes.cpp.obj) : error LNK2001: unresolved external symbol __CxxFrameHandler4
1>cpr.lib(session.cpp.obj) : error LNK2001: unresolved external symbol __CxxFrameHandler4
1>cpr.lib(timeout.cpp.obj) : error LNK2001: unresolved external symbol __CxxFrameHandler4
1>cpr.lib(curlholder.cpp.obj) : error LNK2001: unresolved external symbol __CxxFrameHandler4
1>cpr.lib(session.cpp.obj) : error LNK2001: unresolved external symbol __GSHandlerCheck_EH4
1>cpr.lib(timeout.cpp.obj) : error LNK2001: unresolved external symbol __GSHandlerCheck_EH4
1>cpr.lib(cookies.cpp.obj) : error LNK2001: unresolved external symbol __GSHandlerCheck_EH4
1>cpr.lib(util.cpp.obj) : error LNK2001: unresolved external symbol __GSHandlerCheck_EH4
```

kindly advise.

Thank you.

Rupesh

**Comment by Osyotr** [comment]
vcpkg uses latest available version of msvc on your machine. Do you by any chance have newer version of msvc than the one, that come with vs2015?

**Comment by JonLiu1993** [comment]
@rupeshbhurke ,Can you upgrade the following vs to the latest version, I can install normally under windows10 and vs2019 environment

**Comment by walbourn** [comment]
``__CxxFrameHandler4`` indicates that the library is building with the VS 2019 x64 toolset (19.20 - 19.28), but is being linked against an *older* version of the Visual C++ Runtime. The [binary compatibility](https://docs.microsoft.com/en-us/cpp/porting/binary-compat-2015-2017) story for VS 2015/2017/2019 only works if the final link is done by the 'newest' toolset in the mix. This entry-point is found in ``vcruntime140_1.dll`` which is part of VS 2019 version of the Visual C++ Runtime.

> This entry-point is specifically related to a new x64 exception handler implementation. See [this blog post](https://devblogs.microsoft.com/cppblog/making-cpp-exception-handling-smaller-x64/).

**Comment by rupeshbhurke** [comment]
Is there a way to choose tool set to use while installing packages with vcpkg? The project where I intend to use it is in VS2015. Bringing that project to VS2019 could be a problem as it's pretty big and has lot of dependencies. 

Kindly Advise. 

Thank you. 
Rupesh

**Comment by walbourn** [comment]
Note that you *can* build the entire project in VS 2015, and then link it with VS 2019's version of the CRT and it would work...

That said, I suspect what you want to know is how to get vcpkg to use a different installed toolset...

**Comment by rupeshbhurke** [comment]
> that said, I suspect what you want to know is how to get vcpkg to use a different installed toolset...
Yes, I mean that! Is there a way?

**Comment by rupeshbhurke** [comment]
Hi There,

I solved my issue by using "triplets" feature. I first tried to add my own .cmake files in triplets folder but it kept showing me "Invalid triplet" error. Whenever I tried to use my triplet. So I made changes to existing ones to use toolset v140. It worked as intended.

I was able to use cpr in VS2015 Console Test project. Once I complete testing I will add code to my main project.

Thank you all for the guidance.

With warm regards,
Rupesh
